tests/infrastructure/tls-configuration.go

```diff
@@ -102,10 +99,10 @@ var _ = DescribeInfra("tls configuration", func() {
 					CipherSuites:       kvtls.CipherSuiteIds([]string{cipher.Name}),
 				}
 				conn, err := tls.Dial("tcp", fmt.Sprintf("localhost:844%d", i), acceptedTLSConfig)
-				Expect(err).ToNot(HaveOccurred())
-				Expect(conn).ToNot(BeNil())
-				Expect(conn.ConnectionState().Version).To(BeEquivalentTo(tls.VersionTLS12))
-				Expect(conn.ConnectionState().CipherSuite).To(BeEquivalentTo(cipher.ID))
+				Expect(conn).ToNot(BeNil(), fmt.Sprintf("Should accept valid tls config, %s", err))
+				Expect(err).ToNot(HaveOccurred(), "Should accept valid tls config")
```

Comment:
One liner is enough here (error is asserted implicitly):
```
Expect(tls.Dial("tcp", fmt.Sprintf("localhost:844%d", i), acceptedTLSConfig)).ToNot(BeNil(), fmt.Sprintf("Should accept valid tls config, %s", err))
```

Reply:
Do you suggest that checking the TLS version and used cipher is redundant?

Reply:
I see but in this case, I also need the `conn` to assert the tls&cipher. Therefore I need to leave the call as is and can't use it inside the `Expect`. In theory, I could write a transformer function but that will not make things more readable in this case imho.